### PR TITLE
add LeadingImage component

### PR DESCRIPTION
**Pull request by nickhsine**
- use Image component to render lazy-loading image
- use tiny image as placeholder
- add propTypes and defaultProps for Image and ImageDiff React component
- not to extract heroImage and ogImage from API response to
  redux.state.entities

@garfieldduck @hcchien

### CODE REVIEW

File: src/components/article/Image.js [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
 import { getScreenType } from '../../lib/screen-type'
 import BlockAlignmentWrapper from './BlockAlignmentWrapper'
 import React, { Component } from 'react'
 import LazyLoad from 'react-lazy-load'
 import styles from './Image.scss'
 import UI_SETTING from '../../constants/ui-settings'
 
+const DEFAULT_WIDTH = 200
+const DEFAULT_HEIGHT = 200
+const DEFAULT_PLACEHOLDER = 'https://cdn-images-2.medium.com/freeze/max/30/1*HKrv5OV9P63vz5sa8-Cceg.png?q=20'
```
Comment (hcchien) [inline src/components/article/Image.js]: what is this image? medium.com?

File: src/components/article/Image.js [hunk 2]
```diff
@@ -86,13 +90,14 @@ class Image extends FitwidthMixin(Component) {
 
   render() {
     let imageByDevice = _.get(this.props, [ 'content', 0 ], {})
-    let { mobile, tablet, desktop, original } = imageByDevice
+    let { desktop, tiny } = imageByDevice
```
Comment (hcchien) [inline src/components/article/Image.js]: so we have two sizes only for now?

File: src/components/article/Image.js (hunk 1 above)
Comment (nickhsine) [inline src/components/article/Image.js]: I didn't change the default image placeholder.
May need @garfieldduck  to explain this?

File: src/components/article/LeadingImage.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        leadingImgClass = 'small-leading-img'
+        break
+      default:
+        leadingImgClass = 'leading-img'
+        break
+    }
+
+    if (!image) {
+      return null
+    }
```
Comment (hcchien) [inline src/components/article/LeadingImage.js]: I suggest that you can just move this before the switch/case block.

File: src/components/article/Image.js (hunk 1 above)
Comment (garfieldduck) [inline src/components/article/Image.js]: it'll be replaced soon. there was no tiny image generated by the Keystone api

**Comment by garfieldduck** [comment]
👍 👍 👍  LGTM

File: src/components/article/Image.js (hunk 2 above)
Comment (nickhsine) [inline src/components/article/Image.js]: four sizes including mobile and tablet

**Pull request merged**